src/lazy/expanded/mod.rs

```diff
@@ -524,9 +499,90 @@ impl<Encoding: Decoder, Input: IonInput> ExpandingReader<Encoding, Input> {
         unsafe { &*self.raw_reader.get() }.encoding()
     }
 
+    /// Returns the next IVM, value, or system value as an `ExpandedStreamItem`.
+    ///
+    /// This path is less optimized than `next_system_item` because it needs to surface additional
+    /// items that do not impact the application. However, is useful for tooling that needs more
+    /// visibility into the expansion process.
+    pub fn next_item(&mut self) -> IonResult<ExpandedStreamItem<'_, Encoding>> {
```

Comment:
🗺️ This method can be used by tooling like `ion inspect` to "see" each e-expression before seeing the values that come from evaluating it.

For example, if an e-expression produces no values, the expression itself would never be surfaced by the system reader. Likewise, if an e-expression produced several values, each value would be backed by a template body element--the e-expression wouldn't be accessible.

This alternative view of the expanded layer allows a reader to see all of the expressions _and_ their expansion.